### DISCUSSION TITLE
error 500 on updating product

### DISCUSSION THREAD

**Discussion by zdratoz** [PrestaShop/PrestaShop#34788] [Q&A]
Hallo ,
Im currently facing the following error
Im updating a product and server is responding with error 500  even the changes have been applied

Here is my insomnia call 
![ssss](https://github.com/PrestaShop/PrestaShop/assets/62205792/99c1e0a5-ee9a-4a66-976c-b10df2b132c0)

Any idea?

**Comment by kpodemski** [comment]
Hello @zdratoz 

The error message is quite straightforward. You have an override for the the Search class in override/classes/Search.php. It has some function arguments, but they don't correspond correctly to the original file, which is inside /classes/Search.php.

For example. Original file:

```
public function test($a, $b)
{

}
```

override:

```
public function test($a)
{

}
```

This code won't work. It has to be `$a, $b` in the overridden method.
